XdmodAID.md

```diff
@@ -0,0 +1,853 @@
+                "type": "varchar(64)",
+                "nullable": true,
+                "default": null
+            },
+            {
+                "name": "service_provider",
+                "type": "varchar(64)",
+                "nullable": true,
+                "default": null
+            }
+        "indexes": [{
+            "name": "resource_id",
+            "columns": [
+                "resource_id"
+            ],
+            "is_unique": false
+        }]
+    }
+}
+```
+the modw_cloud database is also formed from the tables defined in
+cloud_common and in cloud_generic.  Each of these directories logical groupings
+of tables when you consider the dataflow.  When data is shredded it is added to
+either the cloud_openstack (in the case of opnstack formatted data), or
+cloud_genderic (in the case of the "generic" format).  During ingestation the
+data flows from cloud_openstack and/or cloud_generic into the cloud_common
+
+The database field sizes are increased using the definitions within the json
+files as opposed using sql to alter the table definition.
+
+There are a similar set of files that seem to define the ETL processes.
+
+##Transaction logs
+
+One thing that I have tried to find and been unsuccessful so far is to get
+transaction logs.  The closest thing that I have hear of is the OpenShift
+audit log, but OpenShift doesn't really enable this in any useful manner by default.
+However for OpenStack, it might be found by processing the rabbit mq messages - not
+entirely sure though.
+
+This is important as it should have the following formt
+```
+Timestamp         | username          | Action
+------------------+-------------------+------------------------------------------------------
+09-AUG-2023 11:18 | <robbaron@bu.edu> | logged on to kaizen (openstack) project: RbbTest (id)
+09-AUG-2023 11:19 | <robbaron@bu.edu> | create cinder volume (vol id)
+09-AUG-2023 11:20 | <robbaron@bu.edu> | created VM (vm id) flavor (...) from volume (vol id)
+09-AUG-2023 11:56 | <robbaron@bu.edu> | logged off from kaizen (openstack) project: RbbTest (id)
+```
+The transaction log should be simple to read and understand, while giving changes
+to the state of the system.  Generally it is ordered by timestamp and have the actions
+of all users on the system.  When explaining things to customers who are paying money, it is
+the easiest and fastest way to point out to the customer why a charge appeared on their invoice.
+
+It is quite painful to explain things to customers without an easy to digest transaction
+log.
+
+##advantages to using xdmod
+```
+  1. community of users
+  2. Seems to have a reeasonable database structure
+  3. does more of what we want to do than cloud forms did
+  4. cloud be a unified platform for collecting and storing metrics on OpenShfit, OpenStack
+     and HPC clusters
+```
+##disavantages to using xdmod
+```
+  1. Not turnkey for cloud services
+  2. No built in support for openshift
```

Comment:
worth running a replace on openstack and openshift to change them to OpenStack & OpenShift to keep things the same.

Reply:
I've done this, but just to clarify, there is the command "xdmod-openstack" which is all lower case and the separate word "OpenStack" - likewise for OpenShift

Reply:
good point, well that's too bad, I am sure you found a smart way to do it like with regex :)